hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ScmSubcommand.java

```diff
@@ -17,21 +17,25 @@
  */
 package org.apache.hadoop.hdds.scm.cli;
 
+import org.apache.hadoop.hdds.scm.cli.upgrade.FinalizeUpgradeBaseCommand;
 import org.apache.hadoop.hdds.scm.client.ScmClient;
 import picocli.CommandLine;
 
 import java.io.IOException;
 import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
 
 /**
  * Base class for admin commands that connect via SCM client.
  */
-public abstract class ScmSubcommand implements Callable<Void> {
+public abstract class ScmSubcommand extends FinalizeUpgradeBaseCommand
```

Comment:
Why do we need ScmSubCommand to extend FinalizeUpgradeBaseCommand? Shouldn't the inheritance be the other way?

Reply:
I changed the structure as discussed over call so this is not applicable anymore.